server/src/main/java/org/tbbtalent/server/api/pub/CvPublicApi.java

```diff
@@ -50,122 +56,144 @@ public CvPublicApi(CandidateService candidateService,
     public Map<String, Object> decodeCvRequest(@PathVariable("token") String token) {
 
         try {
-            String candidateNumber = candidateTokenProvider.getCandidateNumberFromToken(token);
-            Candidate candidate = candidateService.findByCandidateNumber(candidateNumber);
+            CvClaims claims = candidateTokenProvider.getCvClaimsFromToken(token);
+            Candidate candidate = candidateService.findByCandidateNumber(claims.candidateNumber());
```

Comment:
In the non token based approach mentioned earlier, this code would just look up the database from data in the url, returning the candidate number and occupation ids. Then build the CV using your existing CvBuilder